src/common/Gateway.sol

```diff
@@ -362,15 +365,33 @@ contract Gateway is Auth, Recoverable, IGateway {
         transactionRefund = address(0);
     }
 
-    function setRefundAddress(PoolId poolId, address refund) public auth {
+    function _requestPoolFunding(PoolId poolId) internal {
+        IRecoverable refund = subsidy[poolId].refund;
+        if (!poolId.isNull() && address(refund) != address(0)) {
+            uint256 refundBalance = address(refund).balance;
+
+            // Send to the gateway GLOBAL_POT
+            refund.recoverTokens(ETH_ADDRESS, address(this), refundBalance);
+
+            // Extract from the GLOBAL_POT
+            subsidy[GLOBAL_POT].value -= uint96(refundBalance);
+            _subsidizePool(poolId, address(refund), refundBalance);
+        }
+    }
```

Comment:
Finally, this is the request method. Not sure if it's more legible than using the previous `_transientRefundPool`.

1. Add funds to gateway.subsidy[Pool0]
2. Decrement gateway.subsidy[Pool0]
3. Increment gateway.subsidy[PoolA]

Reply:
Hm didn't realize that before, but yeah I do think this is still better than the transient option. Simpler control flow imo.